code/modules/mechs/equipment/combat.dm

```diff
@@ -840,29 +845,34 @@
 		mech.vis_contents.Remove(visual_bluff)
 	QDEL_NULL(visual_bluff)
 
+/obj/item/mech_equipment/shield_generator/installed(mob/living/exosuit/_owner, hardpoint)
+	. = ..()
+	START_PROCESSING(SSobj, src)
```

Comment:
I hate this.

Reply:
Could be better handled by adding a equipment flag for something processing on the mech/Life() instead. (After powered checks have been done)